evalml/tests/pipeline_tests/test_time_series_pipeline.py

```diff
@@ -319,6 +322,36 @@ class Pipeline(pipeline_class):
     pl.score(X, y, objectives)
 
 
+@pytest.mark.parametrize("pipeline_class", [TimeSeriesBinaryClassificationPipeline,
+                                            TimeSeriesMulticlassClassificationPipeline])
+@pytest.mark.parametrize("use_none_X", [True, False])
+def test_score_works_with_estimator_uses_y(use_none_X, pipeline_class, X_y_binary, X_y_multi):
+
+    class Pipeline(pipeline_class):
+        component_graph = [ComponentUsesYInPredict]
+
+    pl = Pipeline({"pipeline": {"gap": 1, "max_delay": 2, "delay_features": False}})
+    if pl.problem_type == ProblemTypes.TIME_SERIES_BINARY:
+        X, y = X_y_binary
+        y = pd.Series(y).map(lambda label: "good" if label == 1 else "bad")
+        expected_unique_values = {"good", "bad"}
+        objectives = ['MCC Binary', "Log Loss Binary"]
+    elif pl.problem_type == ProblemTypes.TIME_SERIES_MULTICLASS:
+        X, y = X_y_multi
+        label_map = {0: "good", 1: "bad", 2: "best"}
+        y = pd.Series(y).map(lambda label: label_map[label])
+        expected_unique_values = {"good", "bad", "best"}
+        objectives = ["MCC Multiclass", "Log Loss Multiclass"]
+
+    if use_none_X:
+        X = None
```

Comment:
Do we already have coverage elsewhere for the case when `use_none_X` is `True`? I.e. do we need `@pytest.mark.parametrize("use_none_X", [True, False])`?

Reply:
Yes I think we do for now. In the other tests with `use_none_X`, we always have a `DelayedFeaturesTransformer` in the pipeline. I think it's important to have coverage when that's not the case. 

That being said, we have coverage for this case for ts regression pipelines in `test_time_series_baseline_regression.py`. In #1580 , I plan on extending the time baseline regressor to classification problems so in that case, I think we can move fold in this coverage into the tests for the baseline.